### DISCUSSION TITLE
Using a VPN and hosting InvenTree remotely (docker)

### DISCUSSION THREAD

**Discussion by ObsoleteMacFan** [inventree/InvenTree#5498] [Q&A]
Good Afternoon,

I am far from a Linux Expert, or any expert in anything really.
I am looking to use InvenTree as an Inventory and Stock Management System for my small business. 
I have read the documentation and was able to install InvenTree bare metal on an old Thinkpad running Ubuntu. immediately when opening localhost on my web browser I was able to log in and have a play around with the software, it looks good, I like it and I want to use it!

Electricity is expensive in the UK and space is at a premium where I live, my business is run from home and eventually I would like a server at home to reduce my reliance on google drive etc. I decided in the short term to purchase a low spec VPS. $10 a year 768mb Ram kind of thing, to do some further testing with InvenTree. 

I managed to install docker via the command line and following the documentation installed the production environment however I am not able to bring up the log in prompt. The server does not have a domain (I do own a domain for the business) I have tried using _my-ip_:8000_and _my-ip_:1337 etc. which appear unresolvable as well as trying some random ports which do bring up a ngix 502 bad gateway error.

is there something simple I am missing with the set up or configuration here? am I correct to assume that the InvenTree system will need to be accessible from the web wether it was stored locally or remotely for the iOS app to work?
I am not adverse to setting up a simple server at home but my better half would be as she is already quite sick of all the random bits of IT laying around the house

I understand this is an incredibly n00b question and I'm sorry for taking up your time but any help would be appreciated. I have searched for a solution but this just isn't clicking into place for me.

Regards
Alan

**Comment by miggland** [comment]
Hi Alan,

as always with such things, there are a string of "simple" issues which could be causing what you see. I'm assuming you don't actually mean VPN, that's something different, but a virtual server.

A virtual server in itself is not a problem to use. You will be able to use the domain, but I would get the IP access working before moving on.

Your first thing is to check connectivity. Have you checked:
- reaching the server by the IP address - SSH, some other service?
- Firewalls?

The message "502 Bad gateway" might be a first good sign, that you're reaching the right server, but the NGINX server isn't reaching InvenTree. You'll need to check the configuration files, and the container logs. Look up the command `docker container  logs` for the next step.

**Reply by ObsoleteMacFan** [reply]
Hey, 
my mistake, I meant VPS! I can reach the server via SSH, Terminal on MacOS. This is how I set up the server. I did install ngix 'bare metal' and was able to go to _ip_ address and it said welcome to ngix so that's all working. From what I read, it appears that docker will override any settings in ufw. I have just reinstalled Ubuntu to give myself a fresh start as i started fiddling with things I don't quite understand.

**Reply by ObsoleteMacFan** [reply]
I have had a look through the inventree-production-inventree-proxy-1 log and can see that on me attempting to connect to port 1337 it has refused it and on attempting to connect to port 8000 it simply says cannot connect to server

**Reply by miggland** [reply]
Probably there's something preventing the Inventree container from starting.

I think docker is a great tool - but it takes some getting used to. If you're not planning on running other services on the same machine, you can also run bare-metal.

**Comment by matmair** [comment]
@ObsoleteMacFan you could also use either package/bare metal on the server.

**Reply by ObsoleteMacFan** [reply]
Perhaps seeing as it is such a low spec server that might be a better idea? I was keen to run it in docker because that would be how I would run it if I had it on a local server.

**Comment by ObsoleteMacFan** [comment]
I have managed to get it running using a bare metal install using the install script! will have a dig around in it to see what was preventing the docker image version
